adapters/repos/db/aggregator/grouper.go

```diff
@@ -224,7 +226,7 @@ func (g *grouper) aggregateAndSelect() ([]group, error) {
 		})
 	}
 
-	return g.topGroups, nil
+	return g.topGroups[g.offset:], nil
```

Comment:
this would probably crash if the length of the results are shorter than the offset. Could you fix that and add a test for this?